lib/fluent/plugin/prometheus.rb

```diff
@@ -165,6 +186,29 @@ def initialize(element, registry, labels)
 
           @base_labels = Fluent::Plugin::Prometheus.parse_labels_elements(element)
           @base_labels = labels.merge(@base_labels)
+
+          @base_initlabels = Fluent::Plugin::Prometheus.parse_initlabels_elements(element)
+        end
+
+        def self.init_label_set(metric, base_initlabels, base_labels)
+           base_initlabels.each { |initlabel|
+             if initlabel.keys.sort != base_labels.keys.sort
+               raise ConfigError, "initlabels for metric #{metric.name} must have the same signature than labels " \
```

Comment:
Good point, no errors showed up during tests so we don't have a test for that?

Don't know how to test easily but this would suffice?

```
      def self.parse_initlabels_elements(conf)
        base_initlabels = []

        conf.elements.select { |e| e.name == 'initlabels' }.each { |block|
          initlabels = {}
          block.each do |key, value|
            # A RecordAccessor has no meaning for initializing a metric's labels
            if value.start_with?('$.') || value.start_with?('$[')
              raise ConfigError, "RecordAccessor cannot be used in an initlabel"
            else 
              initlabels[key.to_sym] = value
            end
          end

          if @multi_worker
            base_labels[:worker_id] = fluentd_worker_id.to_s
          end
          
          base_initlabels << initlabels
        }

        base_initlabels
      end
```

Reply:
> Don't know how to test easily but this would suffice?

Yeah, I think so.